lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1248,11 +1248,7 @@ static bool performCompileStepsPostSILGen(
     return serializeSIB(SM.get(), PSPs, Instance.getASTContext(), MSF);
 
   {
-    const bool haveModulePath = PSPs.haveModuleOrModuleDocOutputPaths();
-    if (haveModulePath && !SM->isSerialized())
-      SM->serialize();
-
-    if (haveModulePath) {
+    if (PSPs.haveModuleOrModuleDocOutputPaths()) {
```

Comment:
Ah, hm. I wonder why this was ever guarding the index stuff. Well, this is definitely what it was doing before, so thank you!

Reply:
cc @benlangmuir